website_customer_also_purchased/model/product.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding:utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C)2010-  OpenERP SA (<http://openerp.com>). All Rights Reserved
+#    App Author: Vauxoo
+#
+#    Developed by Oscar Alcala
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Lesser General Public License as
+#    published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Lesser General Public License for more details.
+#
+#    You should have received a copy of the GNU Lesser General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import osv
+from openerp.osv import fields
+
+
+class product_template(osv.osv):
+    _inherit = 'product.template'
+
+    def _get_purchased(self, cr, uid, ids, field_names, arg=None,
+                       context=None):
+        """
+        This method gets all the products that were purchased in the
+        same sale order og the current product.
+        """
+        result = {}
+        pids_t = []
+        for p_id in ids:
+            result[p_id] = []
+        cr.execute("""
+            select id
+            from product_product
+            where product_tmpl_id = {0}
+            """.format(ids[0]))
+        pid = cr.fetchall()
+        for p in pid:
+            pids_t.append(p[0])
+        pids = '(%s)' % ', '.join(map(repr, tuple(pids_t)))
+        cr.execute("\
```

Comment:
2015-03-24 14:28 GMT-06:00 Olar Alca [Vauxoo] <notifications@github.com>:

> This was done like this to avoid pep8 long lines and to avoid making a sql
> query that cannot be easy to read.


It is not a good reason, I think 1 best SQL is better.

Even thinks on such terms you can create/reuse sale_order_report which
already has part of this SQL, don't you think?


-- 
--------------------
Saludos Cordiales

  [image: --]
Nhomar Hernandez
[image: http://]about.me/nhomar
     <http://about.me/nhomar?promo=email_sig>